### PR TITLE
model done?

### PR DESCRIPTION
**Pull request by etoriaz**

### CODE REVIEW

File: app/models/address.rb [hunk 1]
```diff
@@ -1,2 +1,4 @@
 class Address < ApplicationRecord
+  validate :public_key, presence: true
```
Comment (tlafargue) [inline app/models/address.rb:2]: validates

File: app/models/user.rb [hunk 2]
```diff
@@ -3,4 +3,8 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  validates :username, presence: true, uniqueness: true
+  validates :encrypted_password, presence: true
```
Comment (tlafargue) [inline app/models/user.rb:8]: redundant

File: db/migrate/20230307103545_add_references_to_transactions.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class AddReferencesToTransactions < ActiveRecord::Migration[7.0]
+  def change
```
Comment (tlafargue) [inline db/migrate/20230307103545_add_references_to_transactions.rb:2]: useless

File: db/migrate/20230307104631_add_username_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddUsernameToUsers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :usurname, :string
```
Comment (tlafargue) [inline db/migrate/20230307104631_add_username_to_users.rb:3]: typo usurname

**Review by tlafargue — CHANGES_REQUESTED** [review]

**Review by etoriaz — COMMENTED** [review]
Changes done

**Pull request merged**